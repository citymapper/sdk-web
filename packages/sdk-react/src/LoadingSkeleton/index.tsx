import styled from '../utils/cmStyled'

interface LoadingSkeletonProps {
  width: number | string
  height: number | string
  /**
   * Number of ms to delay the animation
   */
  animationDelay?: number
}

const LoadingSkeleton = styled('div', {
  className: 'cm-loading-skeleton',
})<LoadingSkeletonProps>`
  display: inline-block;
  height: ${({ height }) =>
    typeof height === 'number' ? height + 'px' : height};
  width: ${({ width }) => (typeof width === 'number' ? width + 'px' : width)};
  position: relative;
  overflow: hidden;
  background-color: ${({ theme }) => theme.palette.tone.grey10.color};

  animation: shimmer 3s infinite;
  animation-delay: ${({ animationDelay }) => animationDelay}ms;

  @keyframes shimmer {
    0% {
      opacity: 1;
    }
    8% {
      opacity: 0.5;
    }
    16% {
      opacity: 1;
    }
    100% {
      opacity: 1;
    }
  }

  &.cm-loading-skeleton {
    ${({ theme }) => theme.CmLoadingSkeleton}
  }
`

LoadingSkeleton.defaultProps = {
  animationDelay: 0,
}

export default LoadingSkeleton
