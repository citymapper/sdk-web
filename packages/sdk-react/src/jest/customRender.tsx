import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
