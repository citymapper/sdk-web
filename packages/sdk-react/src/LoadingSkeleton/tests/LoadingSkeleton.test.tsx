import React from 'react'
import customRender from '../../jest/customRender'
import LoadingSkeleton from '../index'

describe('LoadingSkeleton', () => {
  it('matches snapshot', () => {
    const { baseElement } = customRender(
      <LoadingSkeleton height={10} width={100} />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
