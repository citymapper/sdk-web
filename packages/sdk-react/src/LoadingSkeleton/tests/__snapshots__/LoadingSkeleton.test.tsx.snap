// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSkeleton matches snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  height: 10px;
  width: 100px;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

<body>
  <div>
    <div
      class="cm-loading-skeleton emotion-0"
      height="10"
      width="100"
    />
  </div>
</body>
`;
